docs/website/docs/development/architecture/how-odo-works.md

```diff
@@ -0,0 +1,1382 @@
+
+<table>
+<thead>
+<tr>
+<td>Devfile</td>
+<td></td>
+<td>Kubernetes Deployment</td>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>
+
+```yaml
+metadata:
+  # highlight-next-line
+  name: my-sample-java-springboot
+components:
+- name: tools
+  container:
+     image: quay.io/eclipse/che-java11-maven
+     # highlight-start
+     cpuLimit: 500m
+     cpuRequest: 250m
+     memoryLimit: 512Mi
+     memoryRequest: 256Mi
+     # highlight-end
+```
+</td>
+<td> => </td>
+<td>
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+   # highlight-next-line
+   name: my-sample-java-springboot-app
+spec:
+    template:
+        metadata:
+            # highlight-next-line
+            name: my-sample-java-springboot-app
+        spec:
+            containers:
+               - name: tools
+                 # highlight-start
+                 resources:
+                    limits:
+                       cpu: 500m
+                       memory: 512Mi
+                    requests:
+                       cpu: 250m
+                       memory: 256Mi
+                 # highlight-end
+                 image: quay.io/eclipse/che-java11-maven
+                 imagePullPolicy: Always
+```
+</td>
+</tr>
+</tbody>
+</table>
+
+</details>
+
+##### Ports
+
+`odo` translates each element in the `components[].container.endpoints[]` block into a dedicated `containerPort` with the same name and port.
+
+<details>
+<summary>Example</summary>
+
+<table>
+<thead>
+<tr>
+<td>Devfile</td>
+<td></td>
+<td>Kubernetes Deployment</td>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>
+
+```yaml
+metadata:
+  # highlight-next-line
+  name: my-sample-java-springboot
+components:
+- name: tools
+  container:
+     image: quay.io/eclipse/che-java11-maven
+     # highlight-start
+     endpoints:
+     - name: http-springboot
+       targetPort: 8080
+     - name: my-custom-ep
+       targetPort: 3000
+       protocol: http
+       exposure: internal
+     - name: debug
+       targetPort: 5858
+       exposure: none
+     # highlight-end
+```
+</td>
+<td> => </td>
+<td>
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+   # highlight-next-line
+   name: my-sample-java-springboot-app
+spec:
+    template:
+        metadata:
+            # highlight-next-line
+            name: my-sample-java-springboot-app
+        spec:
+            containers:
+               - name: tools
+                 # highlight-start
+                 ports:
+                 - containerPort: 8080
+                   name: http-springboot
+                   protocol: TCP
+                 - containerPort: 3000
+                   name: my-custom-ep
+                   protocol: TCP
+                 - containerPort: 5858
+                   name: debug
+                   protocol: TCP
+                 # highlight-end
+                 image: quay.io/eclipse/che-java11-maven
+                 imagePullPolicy: Always
+```
+</td>
+</tr>
+</tbody>
+</table>
+
+</details>
+
+#### Volumes
```

Comment:
I thin kyou should also tell somewhere in the page that odo can also create `PersistentVolumeClaim` resources when Volume are defined in Devfile or when Ephemeral is false.